packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -1600,95 +1600,270 @@ Update them to point to this script instead?`,
+					],
+				});
+				await mockAUSRequest([]);
+				mockSubDomainRequest();
+				mockUpdateWorkerSubdomain({ enabled: false, previews_enabled: true });
+				mockUploadWorkerRequest({
+					expectedAssets: {
+						jwt: "<<aus-completion-token>>",
+						config: {},
+					},
+					expectedType: "none",
+				});
+				mockPublishRoutesRequest({
+					routes: [
+						// @ts-expect-error - this is what is expected
+						{
+							pattern: "simple.co.uk/path/*",
+						},
+						// @ts-expect-error - this is what is expected
+						{
+							pattern: "simple.co.uk/*",
+						},
+						// @ts-expect-error - this is what is expected
+						{
+							pattern: "*/*",
+						},
+						// @ts-expect-error - this is what is expected
+						{
+							pattern: "*/blog/*",
+						},
+						{
+							pattern: "example.com/blog/*",
+							zone_id: "asdfadsf",
+						},
+						{
+							pattern: "example.com/*",
+							zone_id: "asdfadsf",
+						},
+						{
+							pattern: "example.com/abc/def/*",
+							zone_id: "asdfadsf",
+						},
+					],
+				});
 
-				simple.co.uk/path/*:
-				Workers which have static assets cannot be routed on a URL which has a path component. Update the route to replace /path/* with /*
+				writeWorkerSource();
+				writeAssets([{ filePath: "asset.txt", content: "Content of file-1" }]);
+
+				await runWrangler(`deploy --assets assets`);
 
-				simple.co.uk/:
-				Workers which have static assets must end with a wildcard path. Update the route to end with /*
+				expect(std.warn).toMatchInlineSnapshot(`
+					"[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mWarning: The following routes will attempt to serve Assets on a configured path:[0m
 
-				simple.co.uk:
-				Workers which have static assets must end with a wildcard path. Update the route to end with /*
+					    • simple.co.uk/path/* (Will match assets: assets/path/*)
+					    • */blog/* (Will match assets: assets/blog/*)
+					    • example.com/blog/* (Will match assets: assets/blog/*)
+					    • example.com/abc/def/* (Will match assets: assets/abc/def/*)
 
-				route.co.uk/path:
-				Workers which have static assets cannot be routed on a URL which has a path component. Update the route to replace /path with /*
+					"
+				`);
+				expect(std.out).toMatchInlineSnapshot(`
+					"Total Upload: xx KiB / gzip: xx KiB
+					Worker Startup Time: 100 ms
+					Uploaded test-name (TIMINGS)
+					Deployed test-name triggers (TIMINGS)
+					  simple.co.uk/path/*
+					  simple.co.uk/*
+					  */*
+					  */blog/*
+					  example.com/blog/* (zone id: asdfadsf)
+					  example.com/* (zone id: asdfadsf)
+					  example.com/abc/def/* (zone id: asdfadsf)
+					Current Version ID: Galaxy-Class"
+				`);
+			});
 
-				route.co.uk/path/*:
-				Workers which have static assets cannot be routed on a URL which has a path component. Update the route to replace /path/* with /*
+			it("assets only will not mention invoking Worker on 404", async () => {
```

Comment:
What do you mean by this test title? I'm not picking up its significance.

Reply:
Ah, is this [the `has_user_worker` piece](https://github.com/cloudflare/workers-sdk/pull/7476/files#diff-31fd97cfffe59015a2d5a5cb87902e8e8c016fddd9151455b80129c82567b6a0R231)? I don't see anything checking when that message actually does get added?

Reply:
Yep, apparently I missed that, ty. Added now